snapcraft/tests/test_file_utils.py

```diff
@@ -68,6 +68,25 @@ def test_replace_in_file(self):
                 with open(file_info['path'], 'r') as f:
                     self.assertEqual(f.read(), file_info['expected'])
 
+    def test_replace_in_file_with_permission_error(self):
+        os.makedirs('bin')
+        file_info = {
+            'path': os.path.join('bin', 'readonly'),
+            'contents': '#!/foo/bar/baz/python',
+            'expected': '#!/foo/bar/baz/python',
+        }
+        with open(file_info['path'], 'w') as f:
+            f.write(file_info['contents'])
+
+        with mock.patch('snapcraft.file_utils.open') as mock_open:
+            mock_open.side_effect = PermissionError("")
```

Comment:
I think this could be shortened a little:

```python
with mock.patch('snapcraft.file_utils.open', side_effect=PermissionError):
    # file_utils.replace...
```